### PR TITLE
simplify comparisons

### PR DESCRIPTION
**Pull request by tobymao**
@ginter i got rid of your nulleq simplifications because i deemed they weren't safe

i think mysql nullsafeeq returns 0s and 1s, whereas tsql does true and false

### CODE REVIEW

File: tests/fixtures/optimizer/simplify.sql [hunk 1]
```diff
@@ -372,3 +374,171 @@ CAST('1998-12-01' AS DATE) - INTERVAL '90' foo;
...
+x < 1;
+
+x < 1 OR x < 2;
+x < 1;
+
+x < 2 AND x < 1;
+x < 1;
+
+x < 2 OR x < 1;
+x < 1;
```
Comment (GeorgeSittas) [inline tests/fixtures/optimizer/simplify.sql:433]: Same as above.

File: tests/fixtures/optimizer/simplify.sql [hunk 2]
```diff
@@ -372,3 +374,171 @@ CAST('1998-12-01' AS DATE) - INTERVAL '90' foo;
...
+x <= 1 OR x > 0;
+
+x <= 0 OR x < 0;
+x <= 0;
+
+x >= 0 OR x > 0;
+x >= 0;
+
+x >= 0 OR x > 1;
+x > 1;
```
Comment (GeorgeSittas) [inline tests/fixtures/optimizer/simplify.sql:415]: I think this transformation is invalid. Take `x = 0` as a counterexample.

File: tests/fixtures/optimizer/simplify.sql [hunk 3]
```diff
@@ -372,3 +374,171 @@ CAST('1998-12-01' AS DATE) - INTERVAL '90' foo;
...
+x <= 0 OR x >= 0;
+
+x <= 0 AND x >= 0;
+x <= 0 AND x >= 0;
+
+x < 1 AND x < 2;
+x < 1;
+
+x < 1 OR x < 2;
+x < 1;
```
Comment (GeorgeSittas) [inline tests/fixtures/optimizer/simplify.sql:427]: I think this transformation is invalid. Take `x = 1.5` as a counterexample.

File: tests/fixtures/optimizer/simplify.sql [hunk 4]
```diff
@@ -372,3 +374,171 @@ CAST('1998-12-01' AS DATE) - INTERVAL '90' foo;
...
+
+x < 1 OR x <= 1;
+x <= 1;
+
+x > 1 AND x > 2;
+x > 2;
+
+x > 1 OR x > 2;
+x > 2;
```
Comment (GeorgeSittas) [inline tests/fixtures/optimizer/simplify.sql:457]: Same as above.

File: tests/fixtures/optimizer/simplify.sql [hunk 5]
```diff
@@ -372,3 +374,171 @@ CAST('1998-12-01' AS DATE) - INTERVAL '90' foo;
...
+
+x > 2 AND x > 1;
+x > 2;
+
+x > 2 OR x > 1;
+x > 2;
```
Comment (GeorgeSittas) [inline tests/fixtures/optimizer/simplify.sql:463]: Same as above.

File: tests/fixtures/optimizer/simplify.sql [hunk 6]
```diff
@@ -372,3 +374,171 @@ CAST('1998-12-01' AS DATE) - INTERVAL '90' foo;
...
+x > 1;
+
+x > 1 OR x >= 1;
+x >= 1;
+
+x > 1 AND x >= 2;
+x >= 2;
+
+x > 1 OR x >= 2;
+x >= 2;
```
Comment (GeorgeSittas) [inline tests/fixtures/optimizer/simplify.sql:487]: Same as above.

**Comment by GeorgeSittas** [comment]
Left a few comments, otherwise the PR seems LGTM.

File: sqlglot/helper.py [hunk 7]
```diff
@@ -385,3 +385,13 @@ def dict_depth(d: t.Dict) -> int:
...
+
+
+def first(it: t.Iterable[T]) -> T:
+    """Returns the first element from an iterable.
+
+    Useful for sets.
+    """
+    for i in it:
+        return i
+    raise StopIteration
```
[the remark this answers is not in the document]
Comment (tobymao) [reply sqlglot/helper.py:397]: good idea, will just delete this whole thing, just by doing next(x)

[the remark this answers is not in the document]
Comment (tobymao) [reply sqlglot/helper.py:397]: hm i guess next(set) doesn't work, it takes an iterator

File: tests/fixtures/optimizer/simplify.sql (hunk 6 above)
Comment (tobymao) [reply tests/fixtures/optimizer/simplify.sql]: thanks for catching this

**Pull request merged**